src/libs/DebugUtils.ts

```diff
@@ -0,0 +1,550 @@
+
+    if (typeof array !== 'object' || !Array.isArray(array)) {
+        throw new ArrayError(arrayType);
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    array.forEach((element) => {
+        // Element is an object
+        if (element && typeof element === 'object' && typeof arrayType === 'object') {
+            // eslint-disable-next-line rulesdir/prefer-early-return
+            Object.entries(arrayType).forEach(([key, val]) => {
+                const property = element[key as keyof typeof element];
+                // Property is a constant enum, so we apply validateConstantEnum
+                if (typeof val === 'object' && !Array.isArray(val)) {
+                    validateConstantEnum(property, val as ConstantEnum);
+                }
+                // Expected property type is array
+                if (val === 'array') {
+                    // Property type is not array
+                    if (!Array.isArray(property)) {
+                        throw new ArrayError(arrayType);
+                    }
+                    // Property type is not one of the valid types
+                } else if (Array.isArray(val) ? !val.includes(typeof property) : typeof property !== val) {
+                    throw new ArrayError(arrayType);
+                }
+            });
+            // Element is a constant enum but it's not valid
+        } else if (typeof arrayType === 'object' && !Object.values(arrayType).includes(element)) {
+            throw new ArrayError(arrayType);
+            // Element is not a valid type
+        } else if (typeof element !== arrayType) {
+            throw new ArrayError(arrayType);
+        }
+    });
+}
+
+function validateObject(value: string, type: Record<string, 'string' | 'number' | 'object' | 'array' | 'boolean'>) {
+    if (value === 'undefined') {
+        return;
+    }
+
+    const object = parseJSON(value) as Record<string, 'string' | 'number' | 'object'>;
+
+    if (typeof object !== 'object' || Array.isArray(object)) {
+        throw new ObjectError(type);
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    Object.entries(type).forEach(([key, val]) => {
+        if (val === 'array' ? !Array.isArray(object[key]) : typeof object[key] !== val) {
+            throw new ObjectError(type);
+        }
+    });
+}
+
+function validateString(value: string) {
+    if (value === 'undefined') {
+        return;
+    }
+
+    try {
+        const parsedValue = parseJSON(value);
+
+        if (typeof parsedValue === 'object') {
+            throw new SyntaxError('debug.invalidValue', {cause: {expectedValues: 'string | undefined'}});
+        }
+    } catch (e) {
+        // Only propagate error if value is a string representation of an object or array
+        if ((e as SyntaxError).cause) {
+            throw e;
+        }
+    }
+}
+
+function validateReportDraftProperty(key: keyof Report, value: string) {
```

Comment:
Same, could you explain and put comments about what it's doing? And for the other ones?

Reply:
+1